### PR TITLE
created UserRegistrationDto, SecurityConfig, SecurityInitializer, Val…

### PR DESCRIPTION
**Pull request by Alekearl**
…idators, Exception handler and refactored some classes, plus added new dependencies to pom xml

### CODE REVIEW

File: src/main/java/com/dev/cinema/controller/OrderController.java [hunk 1]
```diff
@@ -36,15 +38,25 @@ public OrderController(OrderService orderService,
...
     @GetMapping
-    public List<OrderDtoResponse> getOrderHistory(@RequestParam Long id) {
-        User user = userService.get(id);
+    public List<OrderDtoResponse> getOrderHistory(@RequestParam Authentication authentication) {
+        Object principal = authentication.getPrincipal();
+        UserDetails userDetails = null;
+        if (principal instanceof UserDetails) {
+            userDetails = (UserDetails) principal;
+        }
+        User user = userService.findByEmail(userDetails.getUsername()).get();
```
Comment (Resci) [inline src/main/java/com/dev/cinema/controller/OrderController.java:50]: Maybe better throw your own exception?

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java [hunk 2]
```diff
@@ -35,15 +37,26 @@ public ShoppingCartController(MovieSessionService movieSessionService,
...
-    public void addMovieSession(@RequestParam Long userId, @RequestParam Long movieSessionId) {
-        User user = userService.get(userId);
+    public void addMovieSession(@RequestParam Authentication authentication,
+                                @RequestParam Long movieSessionId) {
+        Object principal = authentication.getPrincipal();
+        UserDetails userDetails = null;
+        if (principal instanceof UserDetails) {
+            userDetails = (UserDetails) principal;
+        }
+        User user = userService.findByEmail(userDetails.getUsername()).get();
```
Comment (Resci) [inline src/main/java/com/dev/cinema/controller/ShoppingCartController.java:43]: same

File: src/main/java/com/dev/cinema/model/dto/MovieSessionDtoRequest.java [hunk 3]
```diff
@@ -1,8 +1,11 @@
 package com.dev.cinema.model.dto;
 
+import jakarta.validation.constraints.NotNull;
+
 public class MovieSessionDtoRequest {
     private Long movieId;
     private Long cinemaHallId;
```
Comment (Resci) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionDtoRequest.java]: Can we create movie session without movie or cinema hall?

File: src/main/java/com/dev/cinema/model/dto/UserRegistrationDto.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        @FieldsValueMatch(
+                field = "password",
+                fieldMatch = "verifyPassword",
+                message = "Passwords do not match!"
+        ),
+        @FieldsValueMatch(
+                field = "email",
+                fieldMatch = "verifyEmail",
+                message = "Email addresses do not match!"
+        )
```
Comment (Resci) [inline src/main/java/com/dev/cinema/model/dto/UserRegistrationDto.java]: We have 2 field for check email?

**Review by Resci — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/model/dto/MovieSessionDtoRequest.java (hunk 3 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/model/dto/MovieSessionDtoRequest.java]: Changed.

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java (hunk 2 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/controller/ShoppingCartController.java:43]: Seems to me that it was mentioned today. There is no need here in orElseThrow(), we can leave get() here.

File: src/main/java/com/dev/cinema/controller/OrderController.java (hunk 1 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/controller/OrderController.java:50]: Seems to me that it was mentioned today. There is no need here in orElseThrow(), we can leave get() here.

File: src/main/java/com/dev/cinema/model/dto/UserRegistrationDto.java (hunk 4 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/model/dto/UserRegistrationDto.java]: Oh, thank you. 
Changed.

File: src/main/java/com/dev/cinema/model/dto/UserRegistrationDto.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public String getPasswordCheck() {
+        return verifyPassword;
+    }
+
+    public void setPasswordCheck(String passwordCheck) {
+        this.verifyPassword = verifyPassword;
```
Comment (Resci) [inline src/main/java/com/dev/cinema/model/dto/UserRegistrationDto.java]: Incorrect naming

File: src/main/java/com/dev/cinema/validation/annotation/EmailValidator.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        ConstraintValidator<EmailValidatorConstraint, String> {
+    @Override
+    public void initialize(EmailValidatorConstraint emailValidatorConstraint) {
+    }
+
+    @Override
+    public boolean isValid(String contactField,
+                           ConstraintValidatorContext cxt) {
+        return contactField != null && contactField.matches("[0-9]+")
+                && (contactField.length() > 8) && (contactField.length() < 14);
```
Comment (Resci) [inline src/main/java/com/dev/cinema/validation/annotation/EmailValidator.java]: 123456789 is valid email? Why email length may be between 8 and 14?

File: src/main/java/com/dev/cinema/validation/annotation/FieldsValueMatchValidator.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    private String fieldMatch;
+
+    public void initialize(FieldsValueMatch constraintAnnotation) {
+        this.field = constraintAnnotation.field();
+        this.fieldMatch = constraintAnnotation.fieldMatch();
+    }
+
+    public boolean isValid(Object value,
+                           ConstraintValidatorContext context) {
+
```
Comment (Resci) [inline src/main/java/com/dev/cinema/validation/annotation/FieldsValueMatchValidator.java]: empty line

File: src/main/java/com/dev/cinema/validation/annotation/EmailValidator.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
+package com.dev.cinema.validation.annotation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+public class EmailValidator implements
+        ConstraintValidator<EmailValidatorConstraint, String> {
+    @Override
+    public void initialize(EmailValidatorConstraint emailValidatorConstraint) {
+    }
```
Comment (Resci) [inline src/main/java/com/dev/cinema/validation/annotation/EmailValidator.java]: We can remove this method

**Review by Resci — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/validation/annotation/EmailValidator.java (hunk 6 above)
Comment (Resci) [reply src/main/java/com/dev/cinema/validation/annotation/EmailValidator.java]: + var naming

File: src/main/java/com/dev/cinema/model/dto/UserRegistrationDto.java (hunk 5 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/model/dto/UserRegistrationDto.java]: Thank you. =)))

File: src/main/java/com/dev/cinema/validation/annotation/EmailValidator.java (hunk 6 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/validation/annotation/EmailValidator.java]: Changed.

File: src/main/java/com/dev/cinema/validation/annotation/FieldsValueMatchValidator.java (hunk 7 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/validation/annotation/FieldsValueMatchValidator.java]: removed.

File: src/main/java/com/dev/cinema/validation/annotation/EmailValidator.java (hunk 8 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/validation/annotation/EmailValidator.java]: Ok. =)

File: src/main/java/com/dev/cinema/validation/annotation/EmailValidator.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package com.dev.cinema.validation.annotation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+public class EmailValidator implements
+        ConstraintValidator<EmailValidatorConstraint, String> {
+    @Override
+    public boolean isValid(String contactField,
```
Comment (Resci) [inline src/main/java/com/dev/cinema/validation/annotation/EmailValidator.java]: var naming

**Review by Resci — APPROVED** [review]

File: src/main/java/com/dev/cinema/validation/annotation/EmailValidator.java (hunk 9 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/validation/annotation/EmailValidator.java]: changed.

File: src/main/java/com/dev/cinema/config/SecurityConfig.java [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/config/SecurityConfig.java]: we may remove this annotation - @EnableWebSecurity is also a @Configuration

File: src/main/java/com/dev/cinema/config/SecurityConfig.java [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    @Autowired
+    public void configureGlobal(AuthenticationManagerBuilder auth)
+            throws Exception {
+        auth
+                .inMemoryAuthentication()
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/config/SecurityConfig.java:18]: add .passwordEncoder()

File: src/main/java/com/dev/cinema/controller/OrderController.java [hunk 12]
```diff
@@ -36,15 +38,25 @@ public OrderController(OrderService orderService,
     }
 
     @PostMapping("/complete")
-    public void completeOrder(@RequestParam Long id) {
-        User user = userService.get(id);
+    public void completeOrder(@RequestParam Authentication authentication) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/OrderController.java]: do we need @RequestParam?

File: src/main/java/com/dev/cinema/controller/OrderController.java [hunk 13]
```diff
@@ -36,15 +38,25 @@ public OrderController(OrderService orderService,
...
+        Object principal = authentication.getPrincipal();
+        UserDetails userDetails = null;
+        if (principal instanceof UserDetails) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/OrderController.java]: let's remove this check in this case and just cast principal to UserDetails
same for all methods where you get Authentication object

File: src/main/java/com/dev/cinema/controller/OrderController.java [hunk 14]
```diff
@@ -36,15 +38,25 @@ public OrderController(OrderService orderService,
...
+        }
+        User user = userService.findByEmail(userDetails.getUsername()).get();
         ShoppingCart shoppingCart = shoppingCartService.getByUser(user);
         orderService.completeOrder(shoppingCart);
     }
 
     @GetMapping
-    public List<OrderDtoResponse> getOrderHistory(@RequestParam Long id) {
-        User user = userService.get(id);
+    public List<OrderDtoResponse> getOrderHistory(@RequestParam Authentication authentication) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/OrderController.java]: same comment about annotation

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java [hunk 15]
```diff
@@ -35,15 +37,26 @@ public ShoppingCartController(MovieSessionService movieSessionService,
     }
 
     @PostMapping("/movie-sessions")
-    public void addMovieSession(@RequestParam Long userId, @RequestParam Long movieSessionId) {
-        User user = userService.get(userId);
+    public void addMovieSession(@RequestParam Authentication authentication,
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/ShoppingCartController.java]: same

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java [hunk 16]
```diff
@@ -35,15 +37,26 @@ public ShoppingCartController(MovieSessionService movieSessionService,
...
+        }
+        User user = userService.findByEmail(userDetails.getUsername()).get();
         MovieSession movieSession = movieSessionService.get(movieSessionId);
         shoppingCartService.addSession(movieSession, user);
     }
 
     @GetMapping("/by-user")
-    public ShoppingCartDtoResponse getByUser(@RequestParam Long id) {
-        User user = userService.get(id);
+    public ShoppingCartDtoResponse getByUser(@RequestParam Authentication authentication) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/ShoppingCartController.java]: same

File: src/main/java/com/dev/cinema/model/dto/CinemaHallDtoRequest.java [hunk 17]
```diff
@@ -1,7 +1,12 @@
 package com.dev.cinema.model.dto;
 
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.PositiveOrZero;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/CinemaHallDtoRequest.java]: use `javax.validation.constraints` - not jakarta
for all RequestDtos

File: src/main/java/com/dev/cinema/validation/annotation/FieldsValueMatch.java [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Constraint(validatedBy = FieldsValueMatchValidator.class)
+@Target({ ElementType.TYPE })
+@Retention(RetentionPolicy.RUNTIME)
+public @interface FieldsValueMatch {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/validation/annotation/FieldsValueMatch.java]: redundant empty line

File: src/main/java/com/dev/cinema/controller/AuthenticationController.java [hunk 19]
```diff
@@ -19,7 +20,7 @@ public AuthenticationController(AuthenticationService authenticationService) {
     }
 
     @PostMapping("/register")
-    public void registration(@RequestBody UserDtoRequest request) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/AuthenticationController.java:22]: let's remove this class as well if it is not removed yet
`UserDtoRequest`

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/controller/AuthenticationController.java (hunk 19 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/controller/AuthenticationController.java:22]: Removed

File: src/main/java/com/dev/cinema/validation/annotation/FieldsValueMatch.java (hunk 18 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/validation/annotation/FieldsValueMatch.java]: removed.

File: src/main/java/com/dev/cinema/model/dto/CinemaHallDtoRequest.java (hunk 17 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/model/dto/CinemaHallDtoRequest.java]: Searched everywhere and changed.

File: src/main/java/com/dev/cinema/controller/OrderController.java (hunk 13 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/controller/OrderController.java]: Ok.

File: src/main/java/com/dev/cinema/config/SecurityConfig.java (hunk 10 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/config/SecurityConfig.java]: ok. thanx.

File: src/main/java/com/dev/cinema/config/SecurityConfig.java (hunk 11 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/config/SecurityConfig.java:18]: added.

File: src/main/java/com/dev/cinema/controller/OrderController.java (hunk 12 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/controller/OrderController.java]: Nope. =)

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java (hunk 15 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/controller/ShoppingCartController.java]: done

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java (hunk 16 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/controller/ShoppingCartController.java]: done

File: src/main/java/com/dev/cinema/controller/OrderController.java (hunk 14 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/controller/OrderController.java]: done

File: src/main/java/com/dev/cinema/controller/OrderController.java [hunk 20]
```diff
@@ -38,7 +37,7 @@ public OrderController(OrderService orderService,
     }
 
     @PostMapping("/complete")
-    public void completeOrder(@RequestParam Authentication authentication) {
+    public void completeOrder(Authentication authentication) {
         Object principal = authentication.getPrincipal();
         UserDetails userDetails = null;
         if (principal instanceof UserDetails) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/OrderController.java]: was this check removed?

File: src/main/java/com/dev/cinema/model/dto/MovieSessionDtoRequest.java [hunk 21]
```diff
@@ -1,7 +1,7 @@
 package com.dev.cinema.model.dto;
 
-import jakarta.validation.constraints.NotNull;
-import jakarta.validation.constraints.PositiveOrZero;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.PositiveOrZero;
 
 public class MovieSessionDtoRequest {
     @PositiveOrZero
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionDtoRequest.java]: it is not a good idea to allow `0` id for movie or cinemaHall

File: src/main/java/com/dev/cinema/model/dto/UserRegistrationDto.java [hunk 22]
```diff
@@ -2,8 +2,8 @@
 
 import com.dev.cinema.validation.annotation.EmailValidatorConstraint;
 import com.dev.cinema.validation.annotation.FieldsValueMatch;
-import jakarta.validation.constraints.Min;
-import jakarta.validation.constraints.NotNull;
+import javax.validation.constraints.Min;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/UserRegistrationDto.java]: can this annotation be applied to String?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/controller/OrderController.java (hunk 20 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/controller/OrderController.java]: Sorry. ^__^

File: src/main/java/com/dev/cinema/model/dto/MovieSessionDtoRequest.java (hunk 21 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/model/dto/MovieSessionDtoRequest.java]: Ok. Thanx

File: src/main/java/com/dev/cinema/model/dto/UserRegistrationDto.java (hunk 22 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/model/dto/UserRegistrationDto.java]: value of the field or property must be an integer value greater than or equal to the number in the value element - so I`ll remove it and hope won't do this again in a future. =)

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/com/dev/cinema/model/dto/UserRegistrationDto.java (hunk 22 above)
Comment (KseniiaMakarova) [reply src/main/java/com/dev/cinema/model/dto/UserRegistrationDto.java]: you may use `@Size` for String)

**Pull request merged**